engine/src/test/java/org/pentaho/di/trans/steps/http/HTTPTest.java

```diff
@@ -0,0 +1,132 @@
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.reflect.Whitebox.setInternalState;
+
+/**
+ * @author Luis Martins
+ * @since 14-Aug-2018
+ */
+@RunWith( PowerMockRunner.class )
+@PrepareForTest( HttpClientManager.class )
+public class HTTPTest {
+
+  private LogChannelInterface log = mock( LogChannelInterface.class );
+  private RowMetaInterface rmi = mock( RowMetaInterface.class );
+  private HTTPData data = mock( HTTPData.class );
+  private HTTPMeta meta = mock( HTTPMeta.class );
+
+  private final String DATA = "This is the description, there's some HTML here, like &lt;strong&gt;this&lt;/strong&gt;. "
+    + "Sometimes this text is another language that might contain these characters:\n"
+    + "&lt;p&gt;é, è, ô, ç, à, ê, â.&lt;/p&gt; They can, of course, come in uppercase as well: &lt;p&gt;É, È Ô, Ç, À,"
+    + " Ê, Â&lt;/p&gt;. UTF-8 handles this well.";
+
+  @Before
+  public void setup() throws Exception {
+    HttpClientManager.HttpClientBuilderFacade builder = mock( HttpClientManager.HttpClientBuilderFacade.class );
+
+    HttpClientManager manager = mock( HttpClientManager.class );
+    doReturn( builder ).when( manager ).createBuilder();
+
+    CloseableHttpClient client = mock( CloseableHttpClient.class );
+    doReturn( client ).when( builder ).build();
+
+    CloseableHttpResponse response = mock( CloseableHttpResponse.class );
+    doReturn( response ).when( client ).execute( any( HttpGet.class ) );
+
+    BasicHttpEntity entity = new BasicHttpEntity();
+    entity.setContent( new ByteArrayInputStream( DATA.getBytes() ) );
+    doReturn( entity ).when( response ).getEntity();
+
+    mockStatic( HttpClientManager.class );
+    when( HttpClientManager.getInstance() ).thenReturn( manager );
+
+    setInternalState( data, "realUrl", "http://pentaho.com" );
+    setInternalState( data, "argnrs", new int[0] );
+
+    doReturn( false ).when( meta ).isUrlInField();
+    doReturn( "body" ).when( meta ).getFieldName();
+
+    doReturn( false ).when( log ).isDetailed();
+  }
+
+  @Test
+  public void callHttpServiceWithUTF8Encoding() throws Exception {
+    doReturn( "UTF-8" ).when( meta ).getEncoding();
+
+    HTTP http = mock( HTTP.class );
+    setInternalState( http, "log", log );
+    doCallRealMethod().when( http ).callHttpService( any( RowMetaInterface.class ), any( Object[].class ) );
+    doReturn( HttpURLConnection.HTTP_OK ).when( http ).requestStatusCode( any( CloseableHttpResponse.class ) );
+    doReturn( new Header[0] ).when( http ).searchForHeaders( any( CloseableHttpResponse.class ) );
+    setInternalState( http, "data", data );
+    setInternalState( http, "meta", meta );
+
+    Object[] row = http.callHttpService( rmi, new Object[] { 0 } );
+    assertEquals( DATA, row[0] );
+  }
+
+  @Test
+  public void callHttpServiceWithoutEncoding() throws Exception {
+    doReturn( null ).when( meta ).getEncoding();
+
+    HTTP http = mock( HTTP.class );
```

Comment:
**(minor)** The following 9 lines are the exact duplication of lines 105-113.
Evaluate to extract this to a single method.